odo/backends/csv.py

```diff
@@ -224,16 +282,11 @@ def append_dataframe_to_csv(c, df, dshape=None, **kwargs):
             kwargs['encoding'] = encoding
         elif sys.version_info[0] == 2:
             kwargs['mode'] = 'ab' if sys.platform == 'win32' else 'at'
-        f = compressed_open[ext(c.path)](c.path, **kwargs)
-    else:
-        f = c.path
 
-    try:
+    with c.open(mode=kwargs.get('mode', 'a')) as f:
```

Comment:
Which line should I match? I think this is clear that we are passing a default mode